### PR TITLE
Fix a few easy issues

### LINKED ISSUE #704: RepositoryAccessLevel is missing in ProjectUpdate
It looks like other access levels are implemented e.g. `IssuesAccessLeve ` (missing the "l" 😄) or `MergeRequestsAccessLevel ` only `RepositoryAccessLevel` ("repository_access_level")  is missing.

Was the property simply forgotten to be implemented or is something else preventing this access level from being changed via the API?

Thanks

### LINKED ISSUE #705: Reset or Reverting commit
i want an api for resetting or reverting commit on some branch on my repository. does the nuget have this feature? 

https://docs.gitlab.com/ee/api/commits.html#revert-a-commit

### LINKED ISSUE #732: Event options are in lowercase.
EventAction and EventTargetType are in lowercase:
https://docs.gitlab.com/ee/api/events.html#actions

Uppercase query filter does not work.

\NGitLab\Impl\EventClient.cs
image

### LINKED ISSUE #737: MIssing "weight" field in IssueCreate/Edit
https://docs.gitlab.com/ee/api/issues.html#new-issue
https://docs.gitlab.com/ee/api/issues.html#edit-an-issue

### PR DESCRIPTION
**Pull request by louis-z**
- Add `ProjectUpdate.RepositoryAccessLevel`
- Add `ICommitClient.Revert`
- Make sure event action filter parameters are in lowercase _(at least for actions listed [here](https://docs.gitlab.com/ee/user/profile/contributions_calendar.html#user-contribution-events))_
- Add missing `Weight` property to `IssueCreate` and `IssueEdit`
- Slightly improve `EventTests`

Fixes #704, fixes #705, fixes #732, fixes #737

### CODE REVIEW

File: NGitLab/Models/EventAction.cs [hunk 1]
```diff
@@ -5,20 +5,32 @@ namespace NGitLab.Models;
 public enum EventAction
 {
     Accepted,
```
Comment (Toa741) [inline NGitLab/Models/EventAction.cs:8]: Is letting few values without EnumMember intended?

Comment (louis-z) [reply NGitLab/Models/EventAction.cs:8]: It is.

I kept the actions not listed [here](https://docs.gitlab.com/ee/user/profile/contributions_calendar.html#user-contribution-events) unchanged, as I wondered whether they truly are recognized by GitLab. If they are not, we should eventually remove them.

Comment (Toa741) [reply NGitLab/Models/EventAction.cs:8]: Not sure that's the only source of truth, but: https://gitlab.com/gitlab-org/gitlab/-/tree/master/config/events

File: NGitLab.Tests/EventTests.cs [hunk 2]
```diff
@@ -13,33 +13,69 @@ public class EventTests
...
+
+        var currentUserId = context.Client.Users.Current.Id;
+        var userEventClient = context.Client.GetUserEvents(currentUserId);
+
+        // Act
+        var closedEvents = userEventClient.Get(new EventQuery { Action = EventAction.Closed });
+        var issueClosedEvent = closedEvents.SingleOrDefault(e => string.Equals(e.TargetTitle, issueTitle));
+
+        // Assert
+        Assert.That(issueClosedEvent, Is.Not.Null);
```
Comment (Toa741) [inline NGitLab.Tests/EventTests.cs:81]: Test parsed eventaction

Comment (louis-z) [reply NGitLab.Tests/EventTests.cs:81]: Done

**Pull request merged**